src/arrow-util/src/builder.rs

```diff
@@ -228,11 +229,182 @@ fn scalar_to_arrow_datatype(scalar_type: &ScalarType) -> Result<(DataType, Strin
         // arrow::datatypes::IntervalUnit::MonthDayNano is not yet implemented in the arrow parquet writer
         // https://github.com/apache/arrow-rs/blob/0d031cc8aa81296cb1bdfedea7a7cb4ec6aa54ea/parquet/src/arrow/arrow_writer/mod.rs#L859
         // ScalarType::Interval => DataType::Interval(arrow::datatypes::IntervalUnit::DayTime)
+        ScalarType::Array(inner) => {
+            // Postgres / MZ Arrays are weird, since they can be multi-dimensional but this is not
+            // enforced in the type system, so can change per-value.
+            // We use a struct type with two fields - one containing the array elements as a list
+            // and the other containing the number of dimensions the array represents. Since arrays
+            // are not allowed to be ragged, the number of elements in each dimension is the same.
+            let (inner_type, inner_name) = scalar_to_arrow_datatype(inner)?;
+            // TODO: Document these field names in our copy-to-s3 docs
```

Comment:
do we have an issue to add these docs? if not can we add one?

Reply:
yeah I've got an all-encompassing issue here https://github.com/MaterializeInc/materialize/issues/26884 that I'm working on today - I'll be documenting all the non-obvious type transformations we deal with